mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs

```diff
@@ -1791,11 +1795,16 @@ impl<MakerCoin: MmCoin + MakerCoinSwapOpsV2, TakerCoin: MmCoin + TakerCoinSwapOp
 
         let refund_args = RefundFundingSecretArgs {
             funding_tx: &self.taker_funding,
-            time_lock: state_machine.taker_funding_locktime(),
+            funding_time_lock: state_machine.taker_funding_locktime(),
+            payment_time_lock: state_machine.taker_payment_locktime(),
             maker_pubkey: &self.negotiation_data.taker_coin_htlc_pub_from_maker,
             taker_secret: state_machine.taker_secret.as_slice(),
             taker_secret_hash: &secret_hash,
+            maker_secret_hash: &self.negotiation_data.maker_secret_hash,
             swap_contract_address: &None,
+            dex_fee: &state_machine.dex_fee,
+            premium_amount: state_machine.taker_premium.to_decimal(),
+            trading_amount: state_machine.taker_volume.to_decimal(),
```

Comment:
P.S. I am thinking about wrapping the legacy set-price/buy/sell functions to add premium to them, then the total will be premium + volume and we won't need any more changes related to premium in order-matching code. This is the simplest way to do it. But if we need to show the premium to taker and not the whole amount we will need more, or the GUI can show the premium as the amount added to the current price of the ticker since they use the price service. Again, just thinking out loud :)

Reply:
> taker_volume in the above calculations should be taker_amount - taker_premium, please fix it in this PR since it's the right place to do so. 

Sorry could you explain why do we need to subtract premium from amount to get volume?

The below is a `premium` definition from TPU specs
> Also, existing protocol lacks a possibility for Maker to have an additional reward (a.k.a. premium)  for providing the liquidity on the platform. 

I would say that trading volume = provided amount + additional reward.

Reply:
> Sorry could you explain why do we need to subtract premium from amount to get volume?

> I would say that trading volume = provided amount + additional reward.

Let's postpone this to a later implementation/PR until we settle on the approach. It all depends on how we will handle the taker order from the `buy`/`sell` RPC. If premium has a separate parameter in `buy`/`sell` requests, then we don't need to subtract. If [`SellBuyRequest::volume`](https://github.com/KomodoPlatform/komodo-defi-framework/blob/6db5b9fb2debdaac401007bcffd71512474382a3/mm2src/mm2_rpc/src/data/legacy/orders.rs#L14) includes the premium then we need to subtract. Adding the premium to the [`SellBuyRequest::volume`](https://github.com/KomodoPlatform/komodo-defi-framework/blob/6db5b9fb2debdaac401007bcffd71512474382a3/mm2src/mm2_rpc/src/data/legacy/orders.rs#L14) and not as a separate parameters has some benefits as this [balance check](https://github.com/KomodoPlatform/komodo-defi-framework/blob/df6ab98b3b2ee84918bfc0bf1bc03d1e1f219985/mm2src/mm2_main/src/lp_ordermatch.rs#L3817-L3826) won't need changing, and will check the same amount as the check [here](https://github.com/KomodoPlatform/komodo-defi-framework/blob/6a231a3dfa7bc42d5040f1394bbe55e0c27809b0/mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs#L962-L971) in swaps at a later stage.

Reply:
added premium calculation to the issue sub points https://github.com/KomodoPlatform/komodo-defi-framework/issues/1895#issue-1786470565